src-ui/src/app/components/app-frame/app-frame.component.ts

```diff
@@ -94,16 +92,21 @@ export class AppFrameComponent implements OnInit {
   }
 
   closeAll() {
-    this.closeMenu()
-    this.openDocumentsService.closeAll()
-
-    let route = this.activatedRoute.snapshot
-    while (route.firstChild) {
-      route = route.firstChild
-    }
-    if (route.component == DocumentDetailComponent) {
-      this.router.navigate([""])
-    }
+    // user may need to confirm losing unsaved changes
+    this.openDocumentsService.closeAll().subscribe(confirmed => {
```

Comment:
pls clean up the subscriptions or make them auto-complete. e.g. unsubscribe is ngOnDestroy or use takeUntil()